cosign/README.md

```diff
@@ -0,0 +1,79 @@
+
+Verification Workflows  
+
+- The verifier must account for different workflows, such as verifying container images, software binaries, or other artifacts.
+- Each workflow may involve different states of metadata, such as embedded signatures versus detached signatures, and requires customization accordingly.
+- `sigstore-go` library includes a few abstractions to support different use cases, testing, and extensibility:
+
+  - `SignedEntity` - an interface type respresenting a signed message or attestation, with a signature and metadata, implemented by `Bundle`, a type which wraps the `Bundle` type from `protobuf-specs`.
+  - `TrustedMaterial` - an interface type representing a trusted set of keys or certificates for verifying certificates, timestamps, and artifact transparency logs, implemented by `TrustedRoot`
+
+## Scenarios
+
+`sigstore-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources.
+These scenarios can be categorized into following main types and for this library we are going to implement the first four verification for basic user scenarios.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signature using **OIDC/Fulcio**-based trust (no private key needed). | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Environments where signatures are verified with a known public key. |
+| **Blob Verification**             | Verifies detached file signatures (e.g., `.sig` file).                 | Verifying detached signatures for documents, binaries, or standalone files. |
+| **Bundle Verification**           | Verifies a set of files signed together as a **bundle**.               | Ensuring integrity of a collection of files or documents signed as a bundle. |
+| **SCT Verification**              | Verifies **timestamp** of a certificate to ensure it was issued at a specific time. | Verifying certificates with timestamps for **long-term validity** or **audit trails**. |
```

Comment:
We focus on artifacts, don't we?

Reply:
Shall we just make a list including Keyless Verification, Key-Based Verification, Blob Verification, Bundle Verification, Timestamping Verification.